src/App.tsx

```diff
@@ -1,56 +1,120 @@
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useRef, useState,
+} from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
+import { Person } from './types/Person';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>();
+
+  const visiblePerson = useMemo(() => {
+    return peopleFromServer.filter(person => person.name.toLowerCase()
+      .includes(appliedQuery.toLowerCase()));
+  }, [appliedQuery]);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const inputField = useRef<HTMLInputElement>(null);
+
+  document.addEventListener('click', (event: MouseEvent) => {
+    if (event.target !== inputField.current) {
+      setQuery('');
+      setIsFocused(false);
+      inputField.current?.blur();
+    }
```

Comment:
It's should be in `useEffect` hook with `document.removeEventListener`: https://www.pluralsight.com/guides/event-listeners-in-react-components